tests/io/test_data_catalog.py

```diff
@@ -266,6 +266,33 @@ def test_multi_catalog_list(self, multi_catalog):
         assert "abc" in entries
         assert "xyz" in entries
 
+    def test_multi_catalog_list_regex(self, multi_catalog):
+        """Test that regex patterns filter data sets accordingly"""
+        assert multi_catalog.list(regex_search="^a") == ["abc"]
+        assert multi_catalog.list(regex_search="a|x") == ["abc", "xyz"]
+        assert multi_catalog.list(regex_search="^(?!(a|x))") == []
+        assert multi_catalog.list(regex_search="def") == []
+
+    def test_multi_catalog_list_bad_regex(self, multi_catalog):
+        """Test that bad regex is caught accordingly"""
+        pattern = "Invalid pattern regular expression provided"
```

Comment:
Can you test that it displays the bad pattern in the error message? You'll probably need to do some funk regex escaping 🙈

Reply:
Updated